base/linalg/uniformscaling.jl

```diff
@@ -178,7 +178,7 @@ function copy!(A::AbstractMatrix, J::UniformScaling)
     return A
 end
 
-cond{T}(J::UniformScaling{T}) = J.λ ≠ zero(T) ? one(T) : convert(T, Inf)
+cond{T}(J::UniformScaling{T}) = J.λ ≠ zero(T) ? one(real(J.λ)) : oftype(real(J.λ), Inf)
```

Comment:
Isn't this pretty much what you suggested in https://github.com/JuliaLang/julia/pull/19906#discussion_r95039561?